core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -112,18 +112,12 @@ def __init__(self, path: str = None, name: str = None, problem_type: str = None,
         self.path_root = path
         if self.path_root is None:
             path_suffix = self.name
-            if len(self.name) > 0:
-                if self.name[0] != os.path.sep:
-                    path_suffix = os.path.sep + path_suffix
             # TODO: Would be ideal to not create dir, but still track that it is unique. However, this isn't possible to do without a global list of used dirs or using UUID.
             path_cur = setup_outputdir(path=None, create_dir=True, path_suffix=path_suffix)
             self.path_root = path_cur.rsplit(self.path_suffix, 1)[0]
             logger.log(20, f"Warning: No path was specified for model, defaulting to: {self.path_root}")
 
-        # v0.9 FIXME: This is a hack, change so we aren't vulnerable to self.path_root breaking things
-        self.path_root = PathConverter.to_relative(self.path_root)
-
-        self.path = self.create_contexts(self.path_root + self.path_suffix)  # TODO: Make this path a function for consistency.
+        self.path = self.create_contexts(os.path.join(self.path_root, self.path_suffix))  # TODO: Make this path a function for consistency.
```

Comment:
Should we consider storing `path` as a `pathlib.Path` object (as mentioned in #3260)? Might not be worth it if this is a lot work + the current PR already addresses all the path problems, but just wanted to mention this option.

Reply:
I believe @yinweisu tried to do this before, but for some reason we decided to keep it as string. @yinweisu do you recall the exact reason?

Reply:
Yea, the reasoning behind is that there are certain downstream calls that expect the path to be a string instead of a `Path` object. Therefore, in order for it to be transparent, what we need to do would be having a path something like this: `str(Path(path) / "path_suffix")`. This looks verbose and unnecessary to me when `os.path.join` can already accomplish the job.